components/nimbus/src/enrollment.rs

```diff
@@ -972,14 +994,17 @@ fn map_enrollments(enrollments: &[ExperimentEnrollment]) -> HashMap<String, &Exp
     map_enrollments
 }
 
-pub(crate) fn filter_experiments_and_enrollments(
-    experiments: &[Experiment],
+pub(crate) fn filter_experiments_and_enrollments<E>(
+    experiments: &[E],
     enrollments: &[ExperimentEnrollment],
-    filter_fn: fn(&Experiment) -> bool,
-) -> (Vec<Experiment>, Vec<ExperimentEnrollment>) {
-    let experiments: Vec<Experiment> = filter_experiments(experiments, filter_fn);
+    filter_fn: fn(&E) -> bool,
+) -> (Vec<E>, Vec<ExperimentEnrollment>)
+where
+    E: SharedExperimentMetadata + Clone,
+{
+    let experiments: Vec<E> = filter_experiments(experiments, filter_fn);
 
-    let slugs: HashSet<String> = experiments.iter().map(|e| e.slug.clone()).collect();
+    let slugs: HashSet<String> = experiments.iter().map(|e| e.get_slug()).collect();
```

Comment:
OH I also forgot to mention the type change. Instead of the type being `&[Experiment]`, it is now `&[E]` — `where E: SharedExperimentMetadata + Clone`. This is akin to having a generic where the generic implements an interface.

Experiment and EnrolledFeatureConfig both have implementations for the `SharedExperimentMetadata` trait, which allows us to use the same method — `get_slug()` — across each struct that implements that trait.

`Clone` is added there as well because we need to be able to clone/move the object and its members.

Reply:
Happy to chat about this one on a call if you'd like